adyen.php

```diff
@@ -1061,4 +1022,105 @@ private function isSimplePaymentMethod($paymentMethod)
                 )
             );
     }
+
+    /**
+     * @param $params
+     *
+     * @throws \PrestaShop\PrestaShop\Adapter\CoreException
+     */
+    public function hookActionFrontControllerSetMedia($params)
+    {
+        $controller = $this->context->controller;
+        if ($controller->php_self == 'order') {
+            $this->registerAdyenJavascript($controller);
+        }
+    }
+
+    /**
+     * @param FrontController $controller
+     *
+     * @throws \PrestaShop\PrestaShop\Adapter\CoreException
+     */
+    private function registerAdyenJavascript($controller)
+    {
+        /** @var \Adyen\PrestaShop\service\adapter\classes\Controller $controllerAdapter */
+        $controllerAdapter = $this->getService('Adyen\PrestaShop\service\adapter\classes\Controller');
+        $controllerAdapter->setController($controller);
+        if ($this->helper_data->isDemoMode()) {
+            $controllerAdapter->registerJavascript(
+                'adyen-component', // Unique ID
```

Comment:
if this should be a unique ID I would say use: 
```suggestion
                'adyen-checkout-component', // Unique ID
```